cpp/src/quantiles/quantiles_util.hpp

```diff
@@ -46,8 +47,8 @@ CUDF_HOST_DEVICE inline Result linear(T lhs, T rhs, double frac)
   // Underflow may occur when converting int64 to double
   // detail: https://github.com/rapidsai/cudf/issues/1417
 
-  auto dlhs             = static_cast<double>(lhs);
-  auto drhs             = static_cast<double>(rhs);
+  auto dlhs             = convert_to_floating<double>(lhs);
+  auto drhs             = convert_to_floating<double>(rhs);
```

Comment:
I have same question as Mark. Why prefer factory function compared to explicit constructor? (both have 1 argument only).

Reply:
The explicit constructor requires a significant amount of code to live in the header due to device and template requirements. This header is included by most of the code base but there are only a few places that need this particular conversion. We decided it would be better to create factory functions for the uncommon cases.

Reply:
OK, thanks for clarifying.